corehq/apps/userreports/models.py

```diff
@@ -1,79 +1,93 @@
-    StringListProperty,
-)
-from dimagi.ext.jsonobject import JsonObject
 from corehq.apps.cachehq.mixins import (
     CachedCouchDocumentMixin,
-    QuickCachedDocumentMixin,
+    QuickCachedDocumentMixin
+)
+from corehq.apps.userreports.app_manager.data_source_meta import (
+    REPORT_BUILDER_DATA_SOURCE_TYPE_VALUES
 )
 from corehq.apps.userreports.const import (
+    DATA_SOURCE_TYPE_AGGREGATE,
+    DATA_SOURCE_TYPE_STANDARD,
     FILTER_INTERPOLATION_DOC_TYPES,
     UCR_SQL_BACKEND,
-    VALID_REFERENCED_DOC_TYPES,
-    DATA_SOURCE_TYPE_STANDARD, DATA_SOURCE_TYPE_AGGREGATE)
-from corehq.apps.userreports.dbaccessors import get_number_of_report_configs_by_data_source, \
-    get_report_configs_for_domain, get_datasources_for_domain
+    VALID_REFERENCED_DOC_TYPES
+)
+from corehq.apps.userreports.dbaccessors import (
+    get_datasources_for_domain,
+    get_number_of_report_configs_by_data_source,
+    get_report_configs_for_domain
+)
 from corehq.apps.userreports.exceptions import (
     BadSpecError,
     DataSourceConfigurationNotFoundError,
+    DuplicateColumnIdError,
+    InvalidDataSourceType,
     ReportConfigurationNotFoundError,
     StaticDataSourceConfigurationNotFoundError,
-    InvalidDataSourceType, DuplicateColumnIdError)
+    ValidationError,
+)
 from corehq.apps.userreports.expressions.factory import ExpressionFactory
 from corehq.apps.userreports.filters.factory import FilterFactory
-from corehq.apps.userreports.indicators.factory import IndicatorFactory
 from corehq.apps.userreports.indicators import CompoundIndicator
+from corehq.apps.userreports.indicators.factory import IndicatorFactory
+from corehq.apps.userreports.reports.factory import (
+    ChartFactory,
+    ReportColumnFactory,
+    ReportOrderByFactory
+)
 from corehq.apps.userreports.reports.filters.factory import ReportFilterFactory
-from corehq.apps.userreports.reports.factory import ChartFactory, \
-    ReportColumnFactory, ReportOrderByFactory
 from corehq.apps.userreports.reports.filters.specs import FilterSpec
 from corehq.apps.userreports.specs import EvaluationContext, FactoryContext
-from corehq.apps.userreports.util import get_indicator_adapter, get_async_indicator_modify_lock_key
 from corehq.apps.userreports.sql.util import decode_column_name
+from corehq.apps.userreports.util import (
+    get_async_indicator_modify_lock_key,
+    get_indicator_adapter
+)
 from corehq.pillows.utils import get_deleted_doc_types
-from corehq.util.couch import get_document_or_not_found, DocumentNotFound
+from corehq.sql_db.connections import UCR_ENGINE_ID
+from corehq.util.couch import DocumentNotFound, get_document_or_not_found
+from corehq.util.quickcache import quickcache
+from dimagi.ext.couchdbkit import (
+    BooleanProperty,
+    DateTimeProperty,
+    DecimalProperty,
+    DictProperty,
+    Document,
+    DocumentSchema,
+    IntegerProperty,
+    ListProperty,
+    SchemaListProperty,
+    SchemaProperty,
+    StringListProperty,
+    StringProperty
```

Comment:
nit, but whatever script you are running to do this is removing trailing commas. `isort` has an `include_trailing_comma` option which you need to manually set to true.

Reply:
sounds like we need a common config https://github.com/dimagi/commcare-hq/pull/23863

Reply:
https://github.com/dimagi/commcare-hq/pull/23799/commits/b60884ca4f3af00faa73a6957b12d3dfc368afdc